oak_services/src/http/mod.rs

```diff
@@ -73,30 +73,32 @@ impl From<http::header::HeaderMap<http::header::HeaderValue>> for HeaderMap {
     }
 }
 
-impl HeaderMap {
-    /// Flatten the headers into an iterator of tuples (String, http::header::HeaderValue).
-    pub fn iter(
-        &self,
-    ) -> impl Iterator<Item = (http::header::HeaderName, http::header::HeaderValue)> + '_ {
-        self.headers
-            .iter()
+impl std::iter::IntoIterator for HeaderMap {
+    type Item = (http::header::HeaderName, http::header::HeaderValue);
+    type IntoIter = Box<dyn Iterator<Item = (http::header::HeaderName, http::header::HeaderValue)>>;
+
+    /// Convert into an iterator over (http::header::HeaderName, http::header::HeaderValue) tuples.
+    fn into_iter(self) -> Self::IntoIter {
+        let iter = self.headers
+            .into_iter()
             .flat_map(|(name, value)| {
-                value.values.iter().filter_map(move |val| {
-                    let name_value_pair = || -> Result<(http::header::HeaderName, http::header::HeaderValue), OakStatus>  {
+                value.values.into_iter().filter_map(move |val| {
+                    let name_value_pair_closure = || -> Result<(http::header::HeaderName, http::header::HeaderValue), OakStatus>  {
```

Comment:
What is this closure actually doing? I'm a bit confused

Reply:
This is only used to allow using `?` in the enclosing lambda. Otherwise the code would be more verbose.